exercises/robot-simulator/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+// The code below is a stub. Just enough to satisfy the compiler.
+// In order to pass the tests you can add-to or change any of this code.
+
+#[derive(PartialEq, Debug)]
+pub enum Direction {
+    North,
+    East,
+    South,
+    West,
+}
+
+pub struct Robot;
+
+impl Robot {
+    #[allow(unused_variables)]
+    pub fn new(x: isize, y: isize, d: &Direction) -> Self {
+        unimplemented!()
+    }
+
+    pub fn turn_right(self) -> Self {
+        unimplemented!()
+    }
+
+    pub fn turn_left(self) -> Self {
+        unimplemented!()
+    }
+
+    pub fn advance(self) -> Self {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    pub fn instructions(self, instructions: &str) -> Self {
+        unimplemented!()
+    }
+
+    pub fn position(&self) -> (isize, isize) {
+        unimplemented!()
+    }
+
+    pub fn direction(&self) -> &Direction {
```

Comment:
I don't think there's anything in the tests that forces Direction to be Copy. If there is, then I agree that this signature makes more sense. But if not, then I think we should stick with &T